src/components/text-editor/types.ts

```diff
@@ -0,0 +1,16 @@
+import { NodeSpec } from 'prosemirror-model';
+
+/**
+ * @private
+ * set to private to avoid usage while under development
+ */
+export type TextEditorPlugin = {
+    node: NodeConfig[];
+};
+
+export interface NodeConfig {
+    tagName: string;
+    attrs: string[];
+}
+
+export type NodeSpecFactory = (config: NodeConfig) => NodeSpec;
```

Comment:
I'm not sure we need to export this to consumers

Reply:
Good point. We can house it directly within the nodespec factory